recaf-core/src/main/java/me/coley/recaf/code/CommonClassInfo.java

```diff
@@ -43,6 +43,18 @@ public interface CommonClassInfo extends ItemInfo {
 	 */
 	List<MethodInfo> getMethods();
 
+	/**
+	 * @return Class's inner classes.
+	 */
+	List<InnerClassInfo> getInnerClasses();
+
+	/**
+	 * Breadcrumbs of the outer class. For example: A, B for the class C which has B as an outer class.
+	 * This List <strong>MUST</strong> be sorted in order of the outest first. The last element is the outer of the class itself.
+	 * @return breadcrumbs of the outer class.
+	 */
+	List<String> getOuterClassBreadcrumbs();
```

Comment:
Seems to be UI centric data, while this should be strictly a model of the class data.
Breadcrumbds logic can be moved/stored elsewhere.

Reply:
There are 3 ways for getting all outer classes as a breadcrumb:
- getting from inner classes attribute, and then build up the chain (the actual impl)
  the reason why it's only strings not a pair of strings (internal names) and access flags, is because of the root is not there, so it's not possible to get the access flags for it while reading the class.
- splitting by $ and build up the chain (I suppose that's not a good idea because obfuscation can make this difficult?)
- search for all classes in the same package (or whole workspace? I don't know if you can put your inner class/outer class somewhere else) and slowly getting it build up.

So, I don't know if it's a good idea to build the breadcrumbs "externally" from the class read.